updated speech settings panel and adding default values

AKissmehl opened the pull request:
This PR includes the changes from [PR #2063](https://github.com/WEKIT-ECS/MIRAGE-XR/pull/2063), ensuring that the improvements and fixes from that PR are incorporated. In addition to those changes, this PR introduces a rework of the remaining UI components, including the sub-pages. The UI elements have been refactored into prefabs. Default values have also been set for the menu to enhance usability. Moreover, this PR removes the reliance on built-in functions, replacing them with alternative solutions.

Key Changes:

-     Merged changes from [PR #2063](https://github.com/WEKIT-ECS/MIRAGE-XR/pull/2063).
-     Refactored UI components into prefabs.
-     Set default values for UI prefabs to standardize usage.
-     Removed the use of built-in functions, opting for custom solutions.

AKissmehl committed:
Merge branch 'refs/heads/develop' into andreas/2051-update-the-speech-settings-of-the-virtual-instructor-based-on-the-design-review

AKissmehl committed:
Merge branch 'refs/heads/2055-update-ai-manager-and-vi-to-reflect-new-server-routes-and-standardize-naming' into andreas/2051-update-the-speech-settings-of-the-virtual-instructor-based-on-the-design-review

AKissmehl committed:
Add Start method and update prefab

Implemented a Start method in SpeechSettings to set default AI models. Adjusted various prefab properties including font sizes, positions, and size of UI elements.

AKissmehl committed:
Updated some spacing in the prefab

Alvareon committed:
updated speech settings panel

AKissmehl committed:
Merge remote-tracking branch 'refs/remotes/origin/silyasov/fix_virtual_instructor_settings' into 2051-update-the-speech-settings-of-the-virtual-instructor-based-on-the-design-review

# Conflicts:
#	Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/VirtualInstructorEditor.prefab

AKissmehl committed:
Add UI elements and listeners for SpeechSettings

Introduced new UI elements for different views and their respective back buttons in SpeechSettings. Added button listeners to handle view transitions and fixed data update conditions in the Start method. Also updated the prefab to reflect these changes.

AKissmehl committed:
Enable aiManager initialization

Alvareon commented:
missing prefab of header:
<img width="466" alt="Screenshot 2024-08-22 at 21 18 06" src="https://github.com/user-attachments/assets/b25757fd-1e99-41c9-8aa1-a8cc06762630">

AKissmehl committed:
Add new Header prefab and remove redundant content

Introduced a new Header prefab for better UI structuring.

AKissmehl commented:
@fominykh I ran in to a UX Issue. How to Handel that situation, would the header scroll away as well, or should it be covered? 

https://github.com/user-attachments/assets/b4076b28-d7a7-41d8-9c72-127ba6a9f059

fwild commented on Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/SpeechSettings.cs:
```diff
@@ -49,6 +51,97 @@ public class SpeechSettings : MonoBehaviour
+        /// </summary>
+
+        [SerializeField] 
+        private GameObject PromtView; 
```
PromptView

fwild commented on Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/SpeechSettings.cs:
```diff
@@ -49,6 +51,97 @@ public class SpeechSettings : MonoBehaviour
+        private GameObject TTSView; 
+        
+        [SerializeField]
+        private Button  PromtViewBackBtn; 
```
PromptViewBackBtn

AKissmehl committed:
Fix typos and update UI configurations

Corrected typos in variable names from "PromtView" to "PromptView". Modified several UI components and properties in the VirtualInstructorEditor.prefab, including anchoring, font sizes, and object positions to enhance layout and usability.

Alvareon commented:
@AKissMail pls add meta file for the Header.prefab
<img width="624" alt="Screenshot 2024-08-23 at 16 19 52" src="https://github.com/user-attachments/assets/868119e0-f601-4365-8188-b56b66ffe2e1">

AKissmehl committed:
fix a missing sprite in the audio player

AKissmehl committed:
Add meta file for Header prefab in VirtualInstructor

AKissmehl committed:
Add error handling to model updates and new UI elements

Introduced try-catch blocks to handle potential errors in model initialization, enhancing application robustness. Also added new UI elements and adjusted prefab settings for better alignment and functionality.

Alvareon commented:
@AKissMail pls add another meta file:
<img width="695" alt="Screenshot 2024-08-29 at 13 09 32" src="https://github.com/user-attachments/assets/b919d5dd-d7a0-49a6-bd75-32969a454652">

AKissmehl committed:
Add ContentContainer prefab meta file

AKissmehl committed:
Add missing metadata files for VirtualInstructorManager

AKissmehl committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into 2051-update-the-speech-settings-of-the-virtual-instructor-based-on-the-design-review

# Conflicts:
#	Assets/MirageXR/ContentTypes/VirtualInstructor/VirtualInstructorManager.cs.meta
#	Assets/MirageXR/Managers/VirtualInstructorManager/VirtualInstructorDataModel.cs.meta
#	Assets/MirageXR/Managers/VirtualInstructorManager/VirtualInstrutor.cs.meta
#	Assets/MirageXR/Player/Scripts/RootObject.cs
